### ISSUE TITLE
confusing structure of configurations files

### ISSUE THREAD

**Issue by pk16011990** [shopsys/shopsys#1048]
Project base contains:
```
Resources/config/services/commands.yml
Resources/config/services/cron.yml
Resources/config/services/data_fixtures.yml
Resources/config/forms.yml
Resources/config/services.yml
Resources/config/services_test.yml
```

All of them are DI container configurations, but few of them are in directory `services`. I don't know why, I don't know where to look file what I need and what is right place for next new configuration file...

If all af them shoud be in `services` directory, routing configurations shoud be too - I think.

**Comment by PetrHeinz** [comment]
To me it seems that `services(_test).yml` should be in the `config` root directory, but the `forms.yml` should be inside `services` dir - similarly to [the config structure in `shopsys/framework`](https://github.com/shopsys/shopsys/tree/master/packages/framework/src/Resources/config/).

Why do you think that routing should be in the `services` directory? That's not configuration of DI services...

**Comment by vitek-rostislav** [comment]
I would expect even the `services(_test).yml` files to be in `services` directory :slightly_smiling_face:

**Comment by boris-brtan** [comment]
Hi, @PetrHeinz we have came to the agreement @pk16011990 @vitek-rostislav  that all that will be in services folder.
Routing will be in parent folder `config`.
and there will be one point in FAQ.

**Comment by PetrHeinz** [comment]
Having all the files in the `services` directory will prevent you from using wildcard import in the `services.yml` like this:

```yml
imports:
    - { resource: services/*.yml }
```

Keeping the root configs outside of that directory was enabling that to work in similar fashion like the `packages` directory imported from `app/config/config.yml`.

What made you decide this way?

**Comment by boris-brtan** [comment]
consistency where in `project-base/app/config` it is named config.yml and it loads packages subfolder so in services folder it should be services.yml but would does make sense to use some additional folder inside services folder?

**Comment by vitek-rostislav** [comment]
I would vote for putting the `services.yml` file into `services` folder and all the other service-related configs into a new subfolder (e.g. `services/specialized/` or `services/categorized/` ?)

**Comment by PetrHeinz** [comment]
Will you, in the same manner, move `parameters.yml`, `parameters.yml.dist`, `parameters_common.yml`, `parameters_test.yml`, `parameters_test.yml.dist`, `parameters_version.yml`, `parameters_version.yml.dist`, `paths.yml` into directory `parameters` as well? And maybe the same for all routing-related configs?

**Comment by vitek-rostislav** [comment]
hm, makes sense to me, however, I am not working on this issue - the decision is on @boris-brtan :slightly_smiling_face:

**Comment by boris-brtan** [comment]
> hm, makes sense to me, however, I am not working on this issue - the decision is on @boris-brtan

so it makes sense to you @vitek-rostislav that it will be like 
- `config/services`
- `config/routing`
- `config/packages`
or 
- `Resources/config/`
- `Resources/services/`
- `Resources/routing`
?

**Comment by vitek-rostislav** [comment]
To me, 
* `config/services`
* `config/routing`
* `config/parameters`
* ...

**Comment by boris-brtan** [comment]
> To me,
> 
> * `config/services`
> * `config/routing`
> * `config/parameters`
> * ...

is this solution OK for you @pk16011990 @simara-svatopluk @PetrHeinz ?

**Comment by pk16011990** [comment]
I see `config/parameters` what is it? Part of  `services.yml` is parameters, but for me It is part of services (it is overrided parameters for services...) I dont want separate this parameters from services.yml...

**Comment by PetrHeinz** [comment]
@boris-brtan Personally, I would just move the `forms.yml` in to `services`. I don't see reason to change things further without a very good reason and well-thought-out idea.

This step would be easy, and will get the configuration files to less confusing state that could be described and is consistent with service definitions in `shopsys/framework`.

**Comment by boris-brtan** [comment]
hi @pk16011990 would the last suggestion from PetrHeinz solve your issue?

**Comment by pk16011990** [comment]
Nope... It will be still inconsistent, because some definitions of services will be in two different places/directories.

And yes, some changes are easier then another, but for close of this issue must solve whole problem in this issue. Or you can writte "Hey, it is not a problem. It is a wanted state." and close it...

Or do you think that problem is only definition of `forms.yml`? It could be... However for me It is whole DI configuration (including services.yml...) and I wrote it in first comment.

**Comment by boris-brtan** [comment]
hi @vitek-rostislav, seems now that this is not in implementation anymore.

**Comment by pk16011990** [comment]
remove stale... configs are still confusing...

**Comment by vitek-rostislav** [comment]
There are many upvotes for this issue, what do you guys think about it? Are you willing to address it? I think it would need a separate user-story to describe the acceptance criteria...what do you think @malyMiso? :slightly_smiling_face:

**Comment by malyMiso** [comment]
Hi guys, @vitek-rostislav thx for mentioning me here. It really helps me to find this issue 🙂 I like the idea of creating separate US, which can be evaluated and discussed by the team, so we can come up with the best solution to fullfill your needs.
Thx for sharing your thougts, how it should be done, I am pretty sure, that we will be inspired by them. 
Pls. stay tuned, I'll let you know ASAP with new informations.

**Comment by malyMiso** [comment]
Hi guys, @pk16011990, @vitek-rostislav by introducing Symfony Flex there is a new configuration structure. Can you have a look, if it is clearer now ?

**Comment by pk16011990** [comment]
Hi, nope in my opinion. 
Configuration files from first post of this issue are still in same structure.

* `services.yml` is located in config directory - OK
* `crom.yml`, `commands.yml`, `data_fixtures.yml` and `twig.yml` are still in subdirectory `services` - WEIRD but can be accepted
* `forms.yml` is located in config directory - CONFUSING

How can I describe better this problem? 
I don't undestand why the `forms.yml` is in different directory than the `twig.yml`. And why is some more specific services (commands, cron, twig) are located in different directory than the `services.yml`.

I can imagine separated directory `services` for some kind of services groups, but why not `forms.yml`. All these files are services definitions...

**Comment by vitek-rostislav** [comment]
Hi @pk16011990, 
> And why is some more specific services (commands, cron, twig) are located in different directory than the services.yml

The reason behind that is that `services.yml` imports all the other config files from the subdirectory, see https://github.com/shopsys/shopsys/blob/master/packages/framework/src/Resources/config/services.yml#L2

> I don't undestand why the forms.yml is in different directory than the twig.yml. And why is some more specific services (commands, cron, twig) are located in different directory than the services.yml.

It seems to be ok in current master, am I missing something?

**Comment by pk16011990** [comment]
hmmm... Is Flex in master or tag 9.0? And I speak about project-base, not FW... I'm confused...

**Comment by pk16011990** [comment]
and yes, `services.yml` imports all from subdirectory, however my question is "Why is not services.yml located in this subdirectory?" Why some services are in one directory and another services in second directory....?

**Comment by grossmannmartin** [comment]
Stalebot! The structure of config files is still confusing, don't you see?!

**Comment by TomasLudvik** [comment]
Hello @pk16011990 and @grossmannmartin, with Symfony Flex has been all configurations moved to different directories. Is this still issue? If so, can you describe what is confusing to you in current version? Thank you

**Comment by pk16011990** [comment]
> Hi, nope in my opinion.
> Configuration files from first post of this issue are still in same structure.
> 
> * `services.yml` is located in config directory - OK
> * `crom.yml`, `commands.yml`, `data_fixtures.yml` and `twig.yml` are still in subdirectory `services` - WEIRD but can be accepted
> * `forms.yml` is located in config directory - CONFUSING
> 
> How can I describe better this problem?
> I don't undestand why the `forms.yml` is in different directory than the `twig.yml`. And why is some more specific services (commands, cron, twig) are located in different directory than the `services.yml`.
> 
> I can imagine separated directory `services` for some kind of services groups, but why not `forms.yml`. All these files are services definitions...

how or how many times do I have to explain it? Look at config directory in project-base... :roll_eyes:
